test/reactionsystem.jl

```diff
@@ -0,0 +1,132 @@
+    @test isequal(Catalyst.get_states(rs), [s1, s1s2, s2])
+    @test isequal(Catalyst.get_ps(rs), [k1,c1])
+    @named rs = ReactionSystem(MODEL1)
+    isequal(nameof(rs), :rs)
+
+    # Test ODESystem constructor
+    odesys = ODESystem(MODEL1)
+    trueeqs = Equation[Differential(t)(s1) ~ k1]
+    @test isequal(Catalyst.get_eqs(odesys), trueeqs)
+    @test isequal(Catalyst.get_iv(odesys), t)
+    @test isequal(Catalyst.get_states(odesys), [s1])
+    @test isequal(Catalyst.get_ps(odesys), [k1, c1])
+    u0 = [s1 => 1.]
+    par = [k1 => 1., c1 => 2.]
+    @test isequal(odesys.defaults, Dict(append!(u0, par)))  # PL: @Anand: for some reason this does not work with `Catalyst.get_default()`
+    @named odesys = ODESystem(MODEL1)
+    isequal(nameof(odesys), :odesys)
+
+    odesys = ODESystem(sbmlfile)
+    trueeqs = Equation[Differential(t)(s1) ~ -c1 * k1 * 2.0s1 * 2.0s2,
+                       Differential(t)(s1s2) ~ c1 * k1 * 2.0s1 * 2.0s2,
+                       Differential(t)(s2) ~ -c1 * k1 * 2.0s1 * 2.0s2]
+    @test isequal(Catalyst.get_eqs(odesys), trueeqs)
+    @test isequal(Catalyst.get_iv(odesys), t)
+    @test isequal(Catalyst.get_states(odesys), [s1, s1s2, s2])
+    @test isequal(Catalyst.get_ps(odesys), [k1, c1])
+    u0 = [s1 => 2*1., s2 => 2*1., s1s2 => 2*1.]
+    par = [k1 => 1., c1 => 2.]
+    @test isequal(odesys.defaults, Dict(append!(u0, par)))
+    @named odesys = ODESystem(MODEL1)
+    isequal(nameof(odesys), :odesys)
+
+    # Test ODEProblem
+    oprob = ODEProblem(MODEL1, [0., 1.])
+    sol = solve(oprob, Tsit5())
+    @test isapprox(sol.u, [[1.], [2.]])
+
+    @test_nowarn ODEProblem(sbmlfile, [0., 1.])
+
+    # Test checksupport
+
+    # Test make_extensive
+    model = SBML.make_extensive(MODEL2)
+    @test isequal(model.species["s2"].initial_amount, (2., ""))
+    @test isequal(model.species["s2"].initial_concentration, nothing)
+
+    kineticmath2_true = SBML.MathApply("*", SBML.Math[
+        SBML.MathIdent("k1"),
+        SBML.MathApply("*", SBML.Math[
+            SBML.MathVal(2.0),
+            SBML.MathIdent("s2")])
+        ])
+    @test isequal(repr(model.reactions["r2"].kinetic_math), repr(kineticmath2_true))
+    @test model.species["s2"].only_substance_units
+
+    # Test to_initial_amounts
+    model = SBML.to_initial_amounts(MODEL1)
+    @test isequal(model.species["s1"].initial_amount, (1., "substance"))
+    @test isequal(model.species["s1"].initial_concentration, nothing)
+    model = SBML.to_initial_amounts(MODEL2)
+    @test isequal(model.species["s2"].initial_amount, (2., ""))
+    @test isequal(model.species["s2"].initial_concentration, nothing)
+
+    # Test to_extensive_math!
+    model = SBML.to_extensive_math!(MODEL1)
+    @test isequal(model.reactions["r1"].kinetic_math, KINETICMATH1)
+    model = SBML.to_extensive_math!(MODEL2)
+    @test isequal(repr(model.reactions["r2"].kinetic_math), repr(kineticmath2_true))
+    #PL: Todo @Mirek/Anand: Why does this not work without the `repr()`?
```

Comment:
what precise part of the structure fails the comparison?

Reply:
btw why `isequal`? The normal `==` seems more appropriate here to me.

Reply:
> what precise part of the structure fails the comparison?

```
Model to MTK conversions: Test Failed at C:\Users\wolf5212\OneDrive - Nexus365\DPhil Project\Programmes\juliadev\dev\LCSB-BioCore\SBML.jl\test\reactionsystem.jl:90
  Expression: isequal((model.reactions["r2"]).kinetic_math, kineticmath2_true)
   Evaluated: isequal(SBML.MathApply("*", SBML.Math[SBML.MathIdent("k1"), SBML.MathApply("*", SBML.Math[SBML.MathVal{Float64}(2.0), SBML.MathIdent("s2")])]), SBML.MathApply("*", SBML.Math[SBML.MathIdent("k1"), SBML.MathApply("*", SBML.Math[SBML.MathVal{Float64}(2.0), SBML.MathIdent("s2")])]))
```
   
> btw why isequal? The normal == seems more appropriate here to me.

Idk, @anandijain and I have been using it like that. @anandijain did you have a reason?

Reply:
Mk, isequal is the way to compare symbolics, i suggest just messing around w boolean operators in symbolics and it will make sense

Reply:
Thanks. I noticed that when testing Symbolics. Still, the question why the test fails if we do not do `repr` is not clear do me. It is certainly not an elegant way for testing.